lava/submit-to-lava.py

```diff
@@ -196,19 +198,33 @@ def _get_job_info_url(self):
         return job_info_url
 
 
-def repo_revision_list(string):
-    """Parse the string to create a dictionary of repos and revisions."""
-    value = {}
-    # The format shoudl be: repo1-name:sha1,repo2-name:tagname
-    # There is no need to wrap it in a try-except block as argparse will take
-    # care of it in case of Exception
-    for data in string.split(","):
-        repo_name, repo_revision = data.split(":")
-        if not repo_revision:
-            msg = "revision for {} is empty".format(repo_name)
+def key_value_data(string):
+    """Validate the string to be in the form key=value."""
+    if string:
+        key, value = string.split("=")
+        if not (key and value):
+            msg = "{} not in 'key=value' format.".format(string)
             raise argparse.ArgumentTypeError(msg)
-        value[repo_name] = repo_revision
-    return value
+        return {key: value}
+    return {}
+
+
+class StoreDictKeyPair(argparse.Action):
+    """Class for storing key/pair values in a dictionary."""
+
+    def __init__(self, option_strings, dest, nargs=None, **kwargs):
+        """Initialise the class."""
+        self._nargs = nargs
+        super(StoreDictKeyPair, self).__init__(
```

Comment:
Not sure the arguments to `super` are needed, you can usually just say `super().__init__(args)` in python 3 and it does the right thing. This is python 2 syntax.

Reply:
Addressed: 4f113faada95a3b367a375ce7276015218dee1b7